app/functions.php

```diff
@@ -1,22 +1,15 @@
 <?php
 /**
- * Copyright © Magento, Inc. All rights reserved.
- * See COPYING.txt for license details.
+ * @deprecated The global function __() is now loaded via Composer in the Magento Framework, the below require is only
+ *             for backwards compatibility reasons and this file will be removed in a future version
+ * @see        Magento\Framework\Phrase\__function.php
  */
 
-/**
- * Create value-object \Magento\Framework\Phrase
- *
- * @return \Magento\Framework\Phrase
- */
-function __()
-{
-    $argc = func_get_args();
-
-    $text = array_shift($argc);
-    if (!empty($argc) && is_array($argc[0])) {
-        $argc = $argc[0];
+$vendorDir = require VENDOR_PATH;
+if (!function_exists('__')) {
```

Comment:
Why not simply move such logic into `__.php`? It's better to mark `app/functions.php` as deprecated without any changes.